kiwi/builder/container.py

```diff
@@ -69,6 +71,14 @@ def __init__(self, xml_state, target_dir, root_dir):
                 '.', self.requested_container_type, '.tar.xz'
             ]
         )
+        self.base_image = None
+
+        base_image = xml_state.build_type.get_derived_from()
+        if base_image:
+            uri = Uri(base_image, 'images')
+            self.base_image = os.sep.join(
+                [self.root_dir, 'image', os.path.basename(uri.translate())]
+            )
```

Comment:
Sure the variable naming should be improved. Then about the uri.is_remote() check I did not include it here because at this stage the base_image is always local, concretely the file that RootImport stores in `root_dir/image` folder.

Anyway, as you said, I agree to divide the PR in two, it will be easier to review and refine, and part would be for the second one.

Thanks for the comments so far